tests/devices/test_null_qubit.py

```diff
@@ -1397,3 +1398,38 @@ def f(x):
     assert qml.math.allclose(res[0], 0)
     assert qml.math.allclose(res[1], 0)
     assert qml.math.allclose(res[2], jax.numpy.zeros((50, 2)))
+
+
+@pytest.mark.usefixtures("enable_graph_decomposition")
+class TestNullQubitGraphModeExclusive:  # pylint: disable=too-few-public-methods
+    """Tests for NullQubit features that require graph mode enabled.
+    The legacy decomposition mode should not be able to run these tests.
+    NOTE: All tests in this suite will auto-enable graph mode via fixture.
+    """
+
+    def test_insufficient_work_wires_causes_fallback(self):
+        """Test that if a decomposition requires more work wires than available on null.qubit,
+        that decomposition is discarded and fallback is used."""
+
+        class MyNullQubitOp(qml.operation.Operator):  # pylint: disable=too-few-public-methods
+            num_wires = 1
+
+        @qml.register_resources({qml.H: 2})
+        def decomp_fallback(wires):
+            qml.H(wires)
+            qml.H(wires)
+
+        @qml.register_resources({qml.X: 1}, work_wires={"burnable": 5})
+        def decomp_with_work_wire(wires):
+            qml.X(wires)
+
+        qml.add_decomps(MyNullQubitOp, decomp_fallback, decomp_with_work_wire)
+
+        tape = qml.tape.QuantumScript([MyNullQubitOp(0)])
+        dev = qml.device("null.qubit", wires=1)  # Only 1 wire, but decomp needs 5 burnable
+        program = dev.preprocess_transforms()
+        (out_tape,), _ = program([tape])
+
+        assert len(out_tape.operations) == 2
+        assert out_tape.operations[0].name == "Hadamard"
+        assert out_tape.operations[1].name == "Hadamard"
```

Comment:
Can we add a test for an operator with no graph based decomposition outside the gateset, and make sure things still run without error?